lib/identity/client.js

```diff
@@ -155,6 +155,27 @@ class Client extends PartialClient {
     return resultQuery
   }
   /**
+   * getLatestSecrets
+   *
+   * @param {string} secretName   The name of the Secret given by User.
+   * @param {string} secretType   The type of the Secret chosen by User.
+   *
+   *
+   * @return {Promise<Record>}
+   */
+  async getLatestSecret(secretName, secretType) {
+    const request = new Search(true)
+    let searchType = `tozny.secret.${SECRET_UUID}.${secretType}.${secretName}`
+    request.match({ type: searchType })
+    let resultQuery = await this.storage.search(request)
```

Comment:
Since search is sorting these ascending, wouldn't the first secret it returned be the oldest version (not the newest)?

Reply:
https://github.com/tozny/e3dbSearchService/blob/master/clients/elastic_query.go#L134
https://github.com/tozny/api-specs/blob/master/search/swagger.yaml#L241

Reply:
Yeah! my work around was to take the last record returned

Reply:
Prefer updating search as discussed https://tozny.slack.com/archives/G0DPBU8T1/p1612902038062200 to support this use case (we will definitely re-use & benefit from that search capability in other areas)

You're current code is making the assumption that there isn't more than a few hundred versions of the secret, instead of checking to see if it needs to page to get more search results, but if you update search you won't have to worry about that ;-)

Reply:
i read that message wrong the first time, no wonder trying to get descending to work, did not